mateuszbuda_brain-segmentation-pytorch_unet.md

```diff
@@ -0,0 +1,66 @@
+---
+layout: hub_detail
+background-class: hub-background
+body-class: hub
+category: researchers
+title: U-Net
+summary: U-Net with batch normalization for biomedical image segmentation with pretrained weights for abnormality segmentation in brain MRI
+image: TCGA_CS_4944-out.png
+author: mateuszbuda
+tags: [vision]
+github-link: https://github.com/mateuszbuda/brain-segmentation-pytorch
+featured_image_1: unet.png
+accelerator: "cuda-optional"
+---
+
+```python
+import torch
+model = torch.hub.load('mateuszbuda/brain-segmentation-pytorch', 'unet',
+    in_channels=3, out_channels=1, init_features=32, pretrained=True)
+
+```
+
+will load U-Net model pre-trained for abnormality segmentation on a dataset of brain MRI volumes [kaggle.com/mateuszbuda/lgg-mri-segmentation](https://www.kaggle.com/mateuszbuda/lgg-mri-segmentation)
+Pre-trained model requires 3 input channels, 1 output channel, and 32 features in the first layer.
+
+### Model Description
+
+This U-Net model comprises four levels of blocks containing two convolutional layers with batch normalization and ReLU activation function, and one max pooling layer in the encoding part and up-convolutional layers instead in the decoding part.
+The number of convolutional filters in each block is 32, 64, 128, and 256.
+The buttleneck layer has 512 convolutional filters.
+From the encoding layers, skip connections are used to the corresponding layers in the decoding part.
+Input image is a 3-channel brain MRI slice from pre-contrast, FLAIR, and post-contrast sequences, respectively.
+
+### Example
+
+Input images for pre-trained model should have 3 channels and be resized to 256x256 pixels and z-score normalized per volume.
+
+```python
+import numpy as np
+from PIL import Image
+from torchvision import transforms
+
+input_image = Image.open("./images/TCGA_CS_4944-in.png")
```

Comment:
can you please do something similar to https://github.com/pytorch/hub/blame/c997bf2542c6b2888f5a8bfac736ddaba70af8c5/pytorch_vision_googlenet.md#L31-L37

This way, it'll work in the generated Google Colab notebook as well.

Reply:
done